R/LSgameofchance.R

```diff
@@ -15,272 +15,146 @@
+    summaryTable$addRows(list(players = 1, pPoint = probWin[1],   pointsGained = xPoints[1], pA = result[[2]],   pS = result[[1]]))
+    summaryTable$addRows(list(players = 2, pPoint = 1-probWin[1], pointsGained = xPoints[2], pA = 1-result[[2]], pS = 1-result[[1]]))
     
     # fill in the plot
-    if (input$check){ # whether plot CI or not
+    if (options[["CI"]]){ # whether plot CI or not
       
       # Credibility interval (highest posterior density interval)
       SimulResult <- result[[4]] # store the simulated result
-      SimulMatrix <- matrix(0, nrow = 1000, ncol = input$s) # the matrix of samples from posterior distribution based on simulated result
+      SimulMatrix <- matrix(0, nrow = 1000, ncol = nSims) # the matrix of samples from posterior distribution based on simulated result
       
-      for (i in 1:input$s){  
+      for (i in 1:nSims){  
         SimulMatrix[, i] <- rbeta(1000, SimulResult[i]*i+1, i-SimulResult[i]*i+1)
       }
-      CredInt <- apply(SimulMatrix, 2, hdi) # record the credibility interval
+      CredInt <- apply(SimulMatrix, 2, HDInterval::hdi) # record the credibility interval
       y.upper <- CredInt[1,]
       y.lower <- CredInt[2,]
       CIPlot0 <- CIPlot0 + 
-        ggplot2::geom_polygon(ggplot2::aes(x = c(1:input$s,input$s:1), y = c(y.upper, rev(y.lower))), 
+        ggplot2::geom_polygon(ggplot2::aes(x = c(1:nSims,nSims:1), y = c(y.upper, rev(y.lower))), 
                      fill = "lightsteelblue")  # CI
     }
     
-    CIPlot0 <- CIPlot0 + 
-      ggplot2::geom_line(color = "darkred", aes(x = c(1:input$s), y = rep(result[[2]], input$s))) +  # analytical prob
-      ggplot2::geom_line(data= NULL, aes(x = c(1:input$s), y = result[[4]])) # simulated prob
+    CIPlot$plotObject <- jaspGraphs::themeJasp(CIPlot0) + 
+      ggplot2::geom_line(color = "darkred", ggplot2::aes(x = c(1:nSims), y = rep(result[[2]], nSims))) +  # analytical prob
+      ggplot2::geom_line(data= NULL, ggplot2::aes(x = c(1:nSims), y = result[[4]])) # simulated prob
     
-    CIPlot$plotObject <- JASPgraphs::themeJasp(CIPlot0)
     
-    
-  }else if (input$n >= 3 & max(k) < input$t){
+  }else if (nPlayers >= 3 & max(xPoints) < winPoints){
     # output of compare_function2, when there are three or more players
-    result <- compare_function2(k, input$t, p, input$s)
+    result <- compare_function2(xPoints, winPoints, probWin, nSims)
     
     # a vector of analytical p for all players, calculated by switching with player 1
     Analytical_Prob <- vector()
-    for (i in 1:length(p)){
-      k_copy <- replace(k, c(1, i), k[c(i, 1)])
-      p_copy <- replace(p, c(1, i), p[c(i, 1)])
-      Analytical_Prob[i] <- compare_function2(k_copy, input$t, p_copy, input$s)[[2]] 
+    for (i in 1:length(probWin)){
+      k_copy <- replace(xPoints, c(1, i), xPoints[c(i, 1)])
+      p_copy <- replace(probWin, c(1, i), probWin[c(i, 1)])
+      Analytical_Prob[i] <- compare_function2(k_copy, winPoints, p_copy, nSims)[[2]] 
```

Comment:
Could you add more informative names for the compare functions? Also, please use camelCase (I prefer snake_case too but the JASP team decided to banish it.... :(